warehouse/malware/__init__.py

```diff
@@ -9,3 +9,51 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from warehouse import db
+from warehouse.malware import utils
+from warehouse.malware.interfaces import IMalwareCheckService
+
+
+@db.listens_for(db.Session, "after_flush")
+def determine_malware_checks(config, session, flush_context):
+    if not session.new:
+        return
+
+    if not any(
+        [
+            obj.__class__.__name__
+            for obj in session.new
+            if obj.__class__.__name__ in utils.valid_check_types()
+        ]
+    ):
+        return
+
+    malware_checks = session.info.setdefault("warehouse.malware.checks", set())
+    enabled_checks = utils.get_enabled_checks(session)
+    for obj in session.new:
```

Comment:
This assumes all objects in `session.new` are _also_ an object that we run checks on. Should it filter out objects that we don't care about? It seems like we have the potential for a KeyError on the next line when `enabled_checks[obj.__class__.__name__]` is called.

Reply:
Answered below.